### PR TITLE
Allow trusted roles and users to post invites

### LINKED ISSUE #195: Allow Trusted Role to post Invites
In some servers such as Blob Emoji Hub, they want to allow the Trusted level to be able to post any invites (such as the blob server invites) in order to help people out with it.

However, it is not possible currently to post an invite when a whitelist has been created for the Trusted role. GearBot will remove the invite.

So, what I am suggesting in this: Allow the Trusted Role to post invite, but have an option of whether the administrator want to enable or disable it in the first place with that role.

### PR DESCRIPTION
**Pull request by BlackHoleFox**
This PR allows for any user who is listed as a `TRUSTED_USER` or has a `TRUSTED_ROLE` to post invite links without being censored. 

It also had a config switch for this at the admin's choice with: `!configure censortrustedbypass <true/false>`

Disclaimer: I have no idea if this is the optimal solution, but this technically *works* and tweaks can be made to better fit with the rest of the configuration commands.

Resolves #195

### CODE REVIEW

File: GearBot/Cogs/Censor.py [hunk 1]
```diff
@@ -11,6 +11,18 @@
 
 
 async def censor_invite(ctx, code, server_name):
+    # Allow for users with a trusted role, or trusted users, to post invite links
+    if Configuration.get_var(ctx.guild.id, "CENSORING", "ALLOW_TRUSTED_BYPASS") == True:
```
Comment (Tarek1337) [inline GearBot/Cogs/Censor.py]: ``== True`` is not needed.

File: GearBot/Cogs/Serveradmin.py [hunk 2]
```diff
@@ -231,6 +231,28 @@ def __init__(self, bot):
...
         await remove_item(ctx, ServerHolder(server), "invite", list_name="whitelist", config_section="CENSORING")
 
+    @configure.command(name="censortrustedbypass")
+    async def enable_trusted_bypass(self, ctx: commands.Context, enabled_status: bool):
+        config_status = Configuration.get_var(ctx.guild.id, "CENSORING", "ALLOW_TRUSTED_BYPASS")
+        
+        if enabled_status is None:
+            await ctx.send(f"{Translator.translate('missing_arg', ctx, arg='enabled')}")
+
+        if enabled_status == True:
```
Comment (Tarek1337) [inline GearBot/Cogs/Serveradmin.py]: Same here

File: GearBot/Cogs/Serveradmin.py [hunk 3]
```diff
@@ -231,6 +231,28 @@ def __init__(self, bot):
...
+            message = MessageUtils.assemble(ctx, "YES", "features_enabled", count=1) + "Censor Trusted Bypass"
+        else:
+            message = MessageUtils.assemble(ctx, "YES", "features_disabled", count=1) + "Censor Trusted Bypass"
+
+        if enabled_status == config_status:
+            if enabled_status == True:
```
Comment (Tarek1337) [inline GearBot/Cogs/Serveradmin.py]: And here

File: GearBot/Cogs/Censor.py [hunk 4]
```diff
@@ -11,6 +11,18 @@
 
 
 async def censor_invite(ctx, code, server_name):
+    # Allow for users with a trusted role, or trusted users, to post invite links
+    if Configuration.get_var(ctx.guild.id, "CENSORING", "ALLOW_TRUSTED_BYPASS"):
+        trusted_roles = Configuration.get_var(ctx.guild.id, "PERMISSIONS", "TRUSTED_ROLES")
```
Comment (AEnterprise) [inline GearBot/Cogs/Censor.py]: this entire block can be simplified by asking perm_utils if the user is trusted

File: GearBot/Cogs/Serveradmin.py [hunk 5]
```diff
@@ -231,6 +231,24 @@ def __init__(self, bot):
     async def remove_from_whitelist(self, ctx: commands.Context, server:int):
         await remove_item(ctx, ServerHolder(server), "invite", list_name="whitelist", config_section="CENSORING")
 
+    @configure.command(name="censortrustedbypass")
+    async def enable_trusted_bypass(self, ctx: commands.Context, enabled_status: bool):
```
Comment (AEnterprise) [inline GearBot/Cogs/Serveradmin.py]: not a fan of this configure command, mostly because it speaks of this as a "feature", while i normally use that term for things like edit logs and censoring, and use the "configure features enable/disable <feature>" command for those 2, so calling it a feature is a bit confusing

File: GearBot/Cogs/Serveradmin.py [hunk 6]
```diff
@@ -231,6 +231,24 @@ def __init__(self, bot):
...
+        config_status = Configuration.get_var(ctx.guild.id, "CENSORING", "ALLOW_TRUSTED_BYPASS")
+        
+        if enabled_status is None:
```
Comment (AEnterprise) [inline GearBot/Cogs/Serveradmin.py]: why is this here? it shouldn't be possible for this to be none due to you not giving it a default value

File: GearBot/Cogs/Serveradmin.py [hunk 7]
```diff
@@ -231,6 +231,24 @@ def __init__(self, bot):
...
+            await ctx.send(f"{Translator.translate('missing_arg', ctx, arg='enabled')}")
+            return
+
+        enabled_string = "enabled" if enabled_status else "disabled"
+        message = MessageUtils.assemble(ctx, "YES", f"features_{enabled_string}", count=1) + "Censor Trusted Bypass"
+
+        if enabled_status == config_status:
+            message = MessageUtils.assemble(ctx, "NO", f"feature_already_{enabled_string}", count=1) + "Censor Trusted Bypass"
```
Comment (AEnterprise) [inline GearBot/Cogs/Serveradmin.py]: only partially translatable string here

File: GearBot/Cogs/Serveradmin.py [hunk 8]
```diff
@@ -231,6 +231,20 @@ def __init__(self, bot):
     async def remove_from_whitelist(self, ctx: commands.Context, server:int):
         await remove_item(ctx, ServerHolder(server), "invite", list_name="whitelist", config_section="CENSORING")
 
+    @configure.command(name="censortrustedbypass")
+    async def enable_trusted_bypass(self, ctx: commands.Context, enabled_status: bool):
+        config_status = Configuration.get_var(ctx.guild.id, "CENSORING", "ALLOW_TRUSTED_BYPASS")
+
+        enabled_string = "enabled" if enabled_status else "disabled"
```
Comment (AEnterprise) [inline GearBot/Cogs/Serveradmin.py]: this should also be translatable

File: lang/en_US.json [hunk 9]
```diff
@@ -276,6 +276,8 @@
   "automod_ban_failed": "[AUTOBAN FAILURE] AUTOBAN was triggered for {user} (``{user_id}``) but I don't have the BAN_MEMBERS permission required to actually ban them! Ban reason: ``{reason}``",
   "censored_message": "Censored message by {user} (``{user_id}``) in {channel}, blacklisted char sequence ``{sequence}`` is not allowed.\n```{message}```",
   "censor_message_failed": "Failed to censor a message by {user} (``{user_id}``), blacklisted char sequence `{sequence}` is not allowed but I was unable to remove it.\n```{message}\n```<{link}>",
+  "censor_trusted_bypass": "Censor Trusted Bypass has been {status}",
```
Comment (AEnterprise) [inline lang/en_US.json]: minor nitpick: this string doesn't make much sense? doesn't this give the new status? also "censor trusted bypass" might confuse people. how about
``Trusted people will no longer be censored`` and ``Trusted people will now be censored as well`` ?
